src/components/engagement/handlers/set-initial-end-date.handler.ts

```diff
@@ -0,0 +1,85 @@
+import { ServerException } from '../../../common';
+import {
+  DatabaseService,
+  EventsHandler,
+  IEventHandler,
+  ILogger,
+  Logger,
+} from '../../../core';
+import { EngagementStatus } from '../dto';
+import { EngagementService } from '../engagement.service';
+import {
+  InternshipEngagementCreatedEvent,
+  InternshipEngagementUpdatedEvent,
+  LanguageEngagementCreatedEvent,
+  LanguageEngagementUpdatedEvent,
+} from '../events';
+
+type SubscribedEvent =
+  | LanguageEngagementCreatedEvent
+  | LanguageEngagementUpdatedEvent
+  | InternshipEngagementCreatedEvent
+  | InternshipEngagementUpdatedEvent;
+
+@EventsHandler(
+  LanguageEngagementCreatedEvent,
+  LanguageEngagementUpdatedEvent,
+  InternshipEngagementCreatedEvent,
+  InternshipEngagementUpdatedEvent
+)
+export class SetInitialEndDate implements IEventHandler<SubscribedEvent> {
+  constructor(
+    private readonly db: DatabaseService,
+    private readonly engagementService: EngagementService,
+    @Logger('engagement:set-initial-end-date') private readonly logger: ILogger
+  ) {}
+
+  async handle(event: SubscribedEvent) {
+    this.logger.debug('Engagement mutation, set initial end date', {
+      ...event,
+      event: event.constructor.name,
+    });
+
+    try {
+      const shouldUpdateInitialEndDate =
+        event.engagement.status === EngagementStatus.Active &&
+        event.engagement.initialEndDate.value == null &&
+        event.engagement.endDate.value != null;
+
+      if (shouldUpdateInitialEndDate) {
```

Comment:
This could be done outside of the try/catch and negated to return early.

Reply:
I pushed a new commit where I moved the if statement outside of the try/catch block. I believe this resolves this comment, but wanted to check if you would prefer an explicit return.